### PR TITLE
[IA-3348] Azure pricing info in Analyses modals, context bar, cloud environments table

### PR DESCRIPTION
**Pull request by mlilynolting**
Show estimated prices for Azure workspaces.

See also https://github.com/broadinstitute/dsp-scripts/pull/139

### CODE REVIEW

File: src/pages/workspaces/workspace/WorkspaceContainer.js
Comment (cahrens) [inline src/pages/workspaces/workspace/WorkspaceContainer.js]: Note to reviewers: this temporary code will be removed when https://github.com/DataBiosphere/terra-ui/pull/3765 merges

Comment (cahrens) [reply src/pages/workspaces/workspace/WorkspaceContainer.js]: The PR has now merged so this branch needs to be rebased to pick up the changes.

File: src/pages/workspaces/workspace/analysis/cost-utils.js
Comment (n8landolt) [inline src/pages/workspaces/workspace/analysis/cost-utils.js]: I think this calculation is incorrect. getAzureDiskCostEstimate is the complete monthly estimate. However, we're additionally multiplying size to that final figure. We may want to move the size multiplier next to getPersistentDiskPriceForRegionHourly.

Comment (n8landolt) [reply src/pages/workspaces/workspace/analysis/cost-utils.js]: Screenshot 2023-02-07 at 10 13 37 AM
Screenshot 2023-02-07 at 10 13 42 AM

**Review by n8landolt — COMMENTED** [review]
Address my comment and Christina's comment and it's good to go! Once that's done I'll approve.

File: src/pages/workspaces/workspace/WorkspaceContainer.js
Comment (mlilynolting) [reply src/pages/workspaces/workspace/WorkspaceContainer.js]: Done!

File: src/pages/workspaces/workspace/analysis/cost-utils.js
Comment (mlilynolting) [reply src/pages/workspaces/workspace/analysis/cost-utils.js]: Done! added to the GCP case in getPersistentDiskCostHourly - other methods rely on getPdPriceForRegionHourly so I didn't want to change its output.

**Review by n8landolt — APPROVED** [review]

File: src/pages/workspaces/workspace/WorkspaceContainer.js
Comment (cahrens) [reply src/pages/workspaces/workspace/WorkspaceContainer.js]: Great, I have removed Workspaces as a reviewer from this PR because I do not think you have edited any Workspaces-owned files at this point. You should get a second reviewer from the IA team.

**Comment by mlilynolting** [comment]
I broke a unit test - fixing now

**Comment by mlilynolting** [comment]
Tests are fixed!

**Comment by mlilynolting** [comment]
jenkins retest

**Comment by mlilynolting** [comment]
jenkins retest

File: src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.test.js
Comment (msilva-broad) [inline src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.test.js]: an assertion with getAllByText typically asserts the length of the results array to be 0, 1, or the expected count.  As is these will always pass since getAllByText returns an array, which even when empty is "truthy".

File: src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.test.js
Comment (msilva-broad) [inline src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.test.js]: I see a lot of use of the act() wrapper in this PR.... Are they really necessary?  Overuse of act is a common mistake.  Try removing these and seeing if the tests work fine without them.  If some are needed, add a quick comment explaining why.

File: src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js
Comment (msilva-broad) [inline src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js:1007]: just FYI, asMockedFn is really only helpful when used in Typescript land... no harm in using it here though.

File: src/pages/workspaces/workspace/analysis/runtime-common.js
Comment (msilva-broad) [inline src/pages/workspaces/workspace/analysis/runtime-common.js:36]: looks like runtime-common.js has visual components and related bits.... how about we rename it to runtime-common-components.js.

Any small things we can do to get away from large "utils" and "common" dumping grounds the better.

File: src/pages/workspaces/workspace/analysis/cost-utils.test.js
Comment (msilva-broad) [inline src/pages/workspaces/workspace/analysis/cost-utils.test.js:9]: if we get a fast-follow-up PR to improve the unit test coverage of cost-utils.js, I'm ok with getting this incremental PR merged.

File: src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.test.js
Comment (mlilynolting) [reply src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.test.js]: Removing any of them prompts a warning from Jest that React state changes should be wrapped in an act(); it seems that even a plain render operation causes a `setBusy()` call which updates state. In fact there was one place where render was being awaited without an act where I had to *add* an `act()` wrapper to remove the warning. I'm of course open to alternatives but it seems like all of them are needed?

File: src/pages/workspaces/workspace/analysis/cost-utils.test.js
Comment (mlilynolting) [reply src/pages/workspaces/workspace/analysis/cost-utils.test.js:9]: Yes, that's my intention!

File: src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.test.js
Comment (msilva-broad) [reply src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.test.js]: no worries...I just don't want folks seeing this code to think an //Act section always needs an act() wrapper... an easy false correlation.   Please add a comment to these to say something like "wrapping component init-time side-effects with act"

Comment (mlilynolting) [reply src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.test.js]: Wonderful, I will do so! Thanks for clarifying; I'm still learning React+Jest so this is helpful for me

**Review by cahrens — APPROVED** [review]
Approving change to WorkspaceContainer. The PR needs a second IA reviewer to review everything else.

**Review by aherbst-broad — APPROVED** [review]

**Pull request merged**
